### PR TITLE
Sign nupkgs

### PR DESCRIPTION
**Pull request by mmitche**
Sign nupkgs after packing them up

### CODE REVIEW

**Comment by mmitche** [comment]
@brettfo One thing I wasn't sure about was whether the following must also be signed:

```
artifacts\10.1\Microsoft.Portable.FSharp.Core.10.2.0.nupkg
artifacts\4.1\FSharp.Core.4.1.19.nupkg
artifacts\4.3\FSharp.Core.4.3.5.nupkg
```

**Comment by brettfo** [comment]
@mmitche All Microsoft.* and FSharp.* packages should be signed.  The only packages we produce that don't need to be signed are the Testing.* ones.

File: build/config/PackageSignToolData.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        "FSharp.Build.dll",
+        "FSharp.Compiler.Interactive.Settings.dll",
+        "FSharp.Compiler.Private.dll",
+        "FSharp.Core.dll",
+        "fsi.exe"
+    ]
+}
+  
+  
+  /orchestrated-release-2-1/20180417-04/symreader/index.json
```
Comment (brettfo) [inline build/config/PackageSignToolData.json]: Extra line?

File: build/config/PackageSignToolData.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      {
+        "certificate": "NuGet",
+        "strongName": null,
+        "values": [
+          "artifacts\\Microsoft.FSharp.Compiler*.nupkg"
+        ]
+      }
+    ],
+    "exclude": [
+        "artifacts\\*symbols.nupkg",
```
Comment (brettfo) [inline build/config/PackageSignToolData.json]: Let's sign the symbols packages, too, in case we decide to start publishing those.

**Comment by brettfo** [comment]
Once this is in a state that you think it's ready I'd like to do a full internal build before merging.

**Comment by brettfo** [comment]
Changes look good and I see that your internal build passed.  I queued one more that does all build steps just to test it all (this is safe because we gate the actual publishing of packages (when it matters) on the branch name).

**Comment by brettfo** [comment]
Full internal build passed.  Thanks, @mmitche.

**Comment by mmitche** [comment]
@brettfo This need sot be ported to vs15.7, want me to do this now or are you handling it?

**Comment by brettfo** [comment]
Can you do the cherry-pick for me?

**Comment by forki** [comment]
"All Microsoft.* and FSharp.* packages should be signed."

the FSharp.* are still co-owned by the fsharp.org foundation. I think thekey needs to be shared and we need a RFC if we start to sign them /cc @ReedCopsey

**Comment by mmitche** [comment]
@brettfo Do we push those packages to nuget.org?

**Comment by dsyme** [comment]
@forki I think it's ok if FSHarp.Core package is signed by Microsoft - the binary has always been signed, we may as well have the package signed.

**Comment by cartermp** [comment]
https://github.com/fsharp/fslang-design/pull/272

**Comment by forki** [comment]
@dsyme yes signing is ok. @cartermp thanks for the RFC!

**Pull request merged**